src/hera/shared/_type_util.py

```diff
@@ -70,11 +72,14 @@ def get_annotated_metadata(annotation, type_):
     return found
 
 
-def get_workflow_annotation(annotation: Any) -> Optional[Union[Artifact, Parameter]]:
+def get_workflow_annotation(annotation: Any) -> "Optional[Union[Artifact, Parameter]]":
     """If given annotation has Artifact or Parameter metadata, return it.
 
     Note that this function will raise the error when multiple Artifact or Parameter metadata are given.
     """
+    from hera.workflows.artifact import Artifact
+    from hera.workflows.parameter import Parameter
```

Comment:
Why did we move the imports into a function?

Reply:
I think this was from the POC code - I had a circular import at some point. Reverted these changes and tests still working 🚀